src/main/resources/META-INF/rewrite/bouncycastle-jdk15to18.yml

```diff
@@ -0,0 +1,60 @@
+#
+# Copyright 2025 the original author or authors.
+# <p>
+# Licensed under the Moderne Source Available License (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# <p>
+# https://docs.moderne.io/licensing/moderne-source-available-license
+# <p>
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.BouncyCastleFromJdk15OnToJdk15to18
+displayName: Migrate BouncyCastle from jdk15on to jdk15to18 for Java < 8
+description: >-
+  This recipe replaces the Bouncycastle artifacts from jdk15on to jdk15to18. jdk15on isn't maintained anymore and jdk18on is only for Java 8 and above. The jdk15to18 artifact is the up-to-date replacement of the unmaintained jdk15on for java < 8.
+tags:
+  - bouncycastle
+recipeList:
+  - org.openrewrite.java.dependencies.ChangeDependency:
+      oldGroupId: org.bouncycastle
+      oldArtifactId: bcprov-jdk15on
+      newArtifactId: bcprov-jdk15to18
```

Comment:
I think we should also update our recipes for Java 8+ to migrate from `jdk15to18` to `jdk18on`, where those now only migrate `jdk15on`.
https://github.com/openrewrite/rewrite-migrate-java/blob/ec56d0be592c4e95b5fc2eada26c55842b7b0c52/src/main/resources/META-INF/rewrite/bouncycastle-jdk18on.yml#L26-L30